### PR TITLE
sink(ticdc): refactor kafka cluster admin client interface to hide sarama

### LINKED ISSUE #8046: kafka admin client hide sarama

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #8046 

### What is changed and how it works?
* Rewrite the `ClusterAdminClient` interface, and hide all things about the sarama.
* At the application level, hide all sarama calling conventions. 
* Make unit test happy work with the new interface.

#### Tests 

 - Unit test

### Release note 

```release-note
`None`
```

### CODE REVIEW

File: cdc/sink/mq/manager/kafka_manager.go [hunk 1]
```diff
@@ -161,39 +160,21 @@ func (m *kafkaTopicManager) getMetadataOfTopics() ([]*sarama.TopicMetadata, erro
...
 func (m *kafkaTopicManager) waitUntilTopicVisible(topicName string) error {
+	topics := []string{topicName}
 	err := retry.Do(context.Background(), func() error {
 		start := time.Now()
-		topicMetaList, err := m.admin.DescribeTopics([]string{topicName})
+		meta, err := m.admin.GetTopicsMeta(topics, false)
 		if err != nil {
-			log.Error("Kafka admin client describe topic failed",
-				zap.String("topic", topicName),
+			log.Warn(" topic not found, retry it",
```
Comment (sdojjy) [inline cdc/sink/mq/manager/kafka_manager.go:168]: why change the log level to warn?

File: pkg/sink/kafka/admin.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+	"strconv"
+	"strings"
+
+	"github.com/Shopify/sarama"
+	"github.com/pingcap/log"
+	cerror "github.com/pingcap/tiflow/pkg/errors"
+	"go.uber.org/zap"
+)
+
+type admin struct {
```
Comment (sdojjy) [inline pkg/sink/kafka/admin.go:27]: is it better to rename it to saramaAdminClient?

File: pkg/sink/kafka/admin.go [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+		Type:        sarama.BrokerResource,
+		Name:        strconv.Itoa(int(controller)),
+		ConfigNames: []string{configName},
+	})
+	if err != nil {
+		return "", err
+	}
+
+	if len(configEntries) == 0 || configEntries[0].Name != configName {
+		log.Warn("Kafka config item not found", zap.String("configName", configName))
```
Comment (sdojjy) [inline pkg/sink/kafka/admin.go:85]: use Error log level?

**Review by sdojjy — APPROVED** [review]

File: cdc/sink/mq/manager/kafka_manager.go (hunk 1 above)
Comment (3AceShowHand) [reply cdc/sink/mq/manager/kafka_manager.go:168]: it's a retryable error, so change it to warn

We should not use `Error` to frequently, it makes logs horrible.

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**